Pod/Classes/TUSafariActivity.m

```diff
@@ -81,9 +81,13 @@ - (void)prepareWithActivityItems:(NSArray *)activityItems
 
 - (void)performActivity
 {
-	BOOL completed = [[UIApplication sharedApplication] openURL:_URL];
-	
-	[self activityDidFinish:completed];
+    __weak typeof(self) weakSelf = self;
```

Comment:
Shouldn't this actually be a case to keep a strong reference to self? We don't have a retain cycle (the block is only referenced by GCD) and we need to be around to finish the activity. Granted, the system is likely referencing us anyway.